feat: add provider pacts by branch endpoints

YOU54F opened the pull request:
Fixes #619 

Provides the following GET routes

- `/pacts/provider/{provider}/branch/`
  - All pacts for a provider, for any consumers main branch
- `/pacts/provider/{provider}/branch/latest`
  - Latest pact for a provider, for any consumers main branch
- `/pacts/provider/{provider}/branch/{branch}`
  - All pacts for a provider, for any consumers named branch
- `/pacts/provider/{provider}/branch/{branch}/latest`
  - Latest pacts for a provider, for any consumers named branch
- `/pacts/provider/{provider}/consumer/{consumer}/branch`
  - All pacts for a provider, for a named consumers main branch
- `/pacts/provider/{provider}/consumer/{consumer}/branch/latest`
  - Latest pact for a provider, for a named consumers main branch
- `/pacts/provider/{provider}/consumer/{consumer}/branch/{branch}`
  - All pacts for a provider, for a named consumers named branch
- `/pacts/provider/{provider}/consumer/{consumer}/branch/{branch}/latest`
  - Latest pact for a provider, for a named consumers named branch

pahnin commented on lib/pact_broker/pacts/service.rb:
```diff
@@ -95,6 +95,10 @@ def find_latest_pacts_for_provider provider_name, options = {}
       end
 
+      def find_latest_pacts_for_provider_for_branch provider_name, options = {}
+        pact_repository.find_latest_pacts_for_provider provider_name, options[:branch]
```
I think this uses 

https://github.com/pact-foundation/pact_broker/blob/87669c36351df60b4dfecaf8996073fb0382fadc/lib/pact_broker/pacts/repository.rb#L146C25-L146C48

https://github.com/pact-foundation/pact_broker/blob/master/lib/pact_broker/pacts/pact_publication_dataset_module.rb#L235

I am not super well versed with the queries so correct me if I am wrong, but I think this is old way of querying?

https://docs.pact.io/blog/2021/07/04/why-we-are-getting-rid-of-tags

instead should we try to use logic based on the below method ?

https://github.com/pact-foundation/pact_broker/blob/master/lib/pact_broker/pacts/pact_publication_dataset_module.rb#L144

YOU54F replied:
ahh very good spot, appreciate the cookie trail!

YOU54F committed:
feat: add provider pacts by branch endpoints

                "pb:latest-branch-pact-version" =>
                 {
                   href: base_url + "/pacts/provider/{provider}/consumer/{consumer}/branch/{branch}/latest",
                   title: "Latest version of a pact for a given consumer, provider and consumer version branch",
                   templated: false
                 },
                 "pb:branch-pact-versions" =>
                 {
                   href: base_url + "/pacts/provider/{provider}/consumer/{consumer}/branch/{branch}",
                   title: "All versions of a pact for a given consumer, provider and consumer version branch",
                   templated: false
                 },
                "pb:latest-provider-pacts-with-branch" =>
                {
                  href: base_url + "/pacts/provider/{provider}/branch/{branch}/latest",
                  title: "Latest pacts for provider with the specified branch",
                  templated: true
                },
                "pb:provider-pacts-with-branch" =>
                {
                  href: base_url + "/pacts/provider/{provider}/branch/{branch}",
                  title: "All pact versions for the provider with the specified consumer version branch",
                  templated: true
                }
